posts/how-to-commit-code-with-the-git-command-line-interface.md

```diff
@@ -0,0 +1,210 @@
+
+```sh
+git add -A
+```
+
+The above command will add all changes in the entire repository directory to the staging area. We can run this from any subdirectory of the repository.
+
+### Removing changes from the staging area
+
+It’s possible to remove changes from the staging area using the `git restore` command. This is useful if you change your mind about what changes you want to commit or have accidentally added some files that you didn’t intend to add.
+
+The `git restore` command expects a file name, similarly to `git add`, and requires the `--staged` flag to indicate that we intend to unstage a file that is currently staged. Here’s an example:
+
+```sh
+git restore --staged README.md
+```
+
+This will remove `README.md` from the staging area. The changes to the file won’t be lost. They will remain in the working directory.
+
+## Step 3: Checking staged changes before committing
+
+Once we’ve selected the changes that we want to commit, it’s a good idea to check the status of the repository again. This helps us confirm that the changes we’re about to commit are the ones we intend to commit.
+
+Now that we’ve added two files, `README.md` and `getting-started.txt`, we can check the status of the repository by running `git status` again. The output now includes a section called `Changes to be committed`:
+
+```
+Changes to be committed:
+  (use "git restore --staged <file>..." to unstage)
+        modified:   README.md
+        new file:   getting-started.txt
+```
+
+The output confirms that we have two changes in the staging area: `README.md`, which is modified, and `getting-started.txt`, which is new.
+
+If we have any other modified files that we haven’t staged, they will be present in the `Changes not staged for commit` section. These changes won’t be included in the commit and will remain in the working directory after the commit is created.
+
+## Step 4: Creating the commit
+
+To create a commit, we use the `git commit` command, which will perform these tasks:
+
+Create a new commit on top of the latest commit in the current branch.
+Move all staged changes into the new commit.
+Attach the date, author, and message to the new commit.
+Update the current branch to include the new commit.
+Clear the staged changes from the staging area.
+
+Before we can run the command, we need to write a commit message.
+
+### Writing a commit message
+
+The commit message is an important piece of information that we need to provide to the `git commit` command. The commit message is the text that describes the changes that we’re making.
+
+Commit messages are helpful when you’re browsing through the commit history of a project. They show you, at a glance, the sequence of changes made to the project. It’s common to browse commits by looking at the log of each commit’s message, date, and author.
+
+Suppose that we’re committing changes to update the readme file and add a getting-started file. Our commit message could be:
```

Comment:
In documentation, readme is in all caps, as in README. Since the second file is a file name, I would write them both with backticks:

```
... update the `README.md` file and add a `getting-started.txt` file.
```